pkg/transactionBackfiller/backfiller.go

```
@@ -296,3 +312,127 @@ func (t *TransactionBackfiller) ProcessBlock(ctx context.Context, blockNumber ui
 	}
 	return nil
 }
+
+// ProcessBlocksWithAddresses processes logs for specified addresses within a block range.
+// It uses eth_getLogs RPC call to efficiently fetch logs for specific addresses.
+//
+// Parameters:
+//   - ctx: Context for the operation
+//   - queueMessage: The backfill message containing the block range, addresses, and handlers
+//
+// Returns:
+//   - *BackfillerResponse: The result of processing the logs
+func (t *TransactionBackfiller) ProcessBlocksWithAddresses(ctx context.Context, queueMessage *BackfillerMessage) *BackfillerResponse {
+	response := &BackfillerResponse{}
+
+	interestingBlocks, logs, err := t.fetcher.FetchInterestingBlocksAndLogsForContractsForBlockRange(ctx, queueMessage.StartBlock, queueMessage.EndBlock, queueMessage.Addresses)
+	if err != nil {
+		t.logger.Sugar().Errorw("Error fetching interesting blocks and logs", zap.Error(err))
+		return response
```

Rather than returning an empty struct, I would return the error